lib/logger.c

```diff
@@ -313,6 +313,11 @@ grn_logger_output_end(grn_ctx *ctx,
         output->file != stderr &&
         (output->rotate_threshold_size > 0 &&
          output->size >= output->rotate_threshold_size)) {
+#ifndef _WIN32
+      if (output->need_flock) {
```

Comment:
Is `&& output->file` needless in this condition?

Because if we don't check `output->file != NULL` in this condition, I think that Groonga will crash at [`flock(fileno(output->file), LOCK_UN)`](https://github.com/groonga/groonga/pull/1871/files#diff-4300374c82a4008ddcc5c3784fbba41e8f8582ff220cabb9ea8581445b9d95a3R318).

Reply:
I see, it seems safe to do a `NULL` check.
It seems that `fclose()` also causes a crash, so I think the fix would be something like this.

```c
if (output->file != NULL) {
#ifndef _WIN32
  if (output->need_flock) {
    flock(fileno(output->file), LOCK_UN);
  }
#endif
  fclose(output->file);
  output->file = NULL;
}
```

Reply:
Is `output->file == NULL` happen here?

Reply:
It should not be `NULL` here.
I thought it would be safe to check just in case, but is it unnecessary?

Reply:
I don't want to add a needless check to keep code simple as much as possible.

Reply:
I got it

@komainu8  I decided not to add the check since it is not `NULL` here.

Reply:
I see.
I agree with it.